apps/test/netsim/NetSimLocalClientNode.js

```diff
@@ -38,6 +38,40 @@ describe("NetSimLocalClientNode", function () {
     assert(undefined !== testRemoteNode, "Made a remote node");
   });
 
+  describe ("onNodeTableChange_", function () {
+    beforeEach(function () {
+      testLocalNode.initializeSimulation(null, null);
+    });
+
+    it ("detects when own row has gone away and calls lost connection callback", function () {
+      var lostConnection = false;
+      testLocalNode.setLostConnectionCallback(function () {
+        lostConnection = true;
+      });
+      assertEqual(false, lostConnection);
+
+      testShard.nodeTable.api_.remoteTable.deleteMany([testLocalNode.entityID], function () {});
+      testShard.nodeTable.refresh();
+      assertEqual(true, lostConnection);
+    });
+
+    it ("also detects if row uuid changes", function () {
+      var lostConnection = false;
+      testLocalNode.setLostConnectionCallback(function () {
+        lostConnection = true;
+      });
+      assertEqual(false, lostConnection);
+
+      // Reset fake remote table and repopulate first two rows.
```

Comment:
I'm renaming the test as you suggested to be more descriptive of what's going on and extracted some common setup.  Since our test API is a pretty good facsimile of the real thing, the only legitimate way to change a UUID _is_ to reset the shard.  We could introduce a test method to -just- change the UUID of a row, but I'm not sure there's a benefit to doing that in addition to the "shard reset and reclaimed ID case."  What do you think?